charmcraft.yaml

```diff
@@ -10,25 +32,75 @@ bases:
       - name: ubuntu
         channel: "22.04"
         architectures: [amd64]
-      - name: centos
-        channel: "7"
-        architectures: [amd64]
 
 parts:
   charm:
-    build-packages: [git]
-    charm-python-packages: [setuptools]
-
-  # Create a version file and pack it into the charm. This is dynamically generated
-  # as part of the build process for a charm to ensure that the git revision of the
-  # charm is always recorded in this version file.
-  version-file:
-    plugin: nil
     build-packages:
-      - git
+      - wget
     override-build: |
-      VERSION=$(git -C $CRAFT_PART_SRC/../../charm/src describe --dirty --always)
-      echo "Setting version to $VERSION"
-      echo $VERSION > $CRAFT_PART_INSTALL/version
-    stage:
-      - version
+       wget https://github.com/mej/nhc/releases/download/1.4.3/lbnl-nhc-1.4.3.tar.gz
+       craftctl default
+
+requires:
+  fluentbit:
+    interface: fluentbit
+
+provides:
+  slurmctld:
+    interface: slurmd
+    limit: 1
+
+config:
+  options:
+    partition-config:
+      type: string
+      default: ""
+      description: >
+        Additional partition configuration parameters, specified as a space separated `key=value`
+        in a single line. Find a list of all possible partition configuration parameters
+        [here](https://slurm.schedmd.com/slurm.conf.html#SECTION_PARTITION-CONFIGURATION).
+        
+  
+        Example usage:
+        ```bash
+         $ juju config slurmd partition-config="DefaultTime=45:00 MaxTime=1:00:00"
+        ```
```

Comment:
Want to know your thoughts on this @jamesbeedy 

One thing that I'm thinking about is that we're potentially different API for configuring partitions v.s. configuring nodes individually. I'm wondering if it would be better to have this be an action rather than a config option? We're already checking if the node is the leader when we update the config in this case, so I don't see it to be much of an issue to have this action only be runnable against the leader:

```
juju run --quiet slurmd/leader set-partition-config parameters="..."  
```

This way we can then have a consistent API for managing the configuration of both partitions and nodes:

```
juju run --quiet slurmd/leader set-partition-config parameters="..."
juju run slurmd/leader show-partition-config
juju run --quiet slurmd/7 set-node-config parameters="..."
juju run slurmd/7 show-node-config
```

Let me know what you think

Reply:
Love it. +1